src/common/src/main/scala/osmesa/common/sources/ChangeSource.scala

```diff
@@ -34,20 +34,15 @@ object ChangeSource extends Logging {
         Thread.sleep(Delay.toMillis)
         getSequence(baseURI, sequence)
       } else {
-        // NOTE: if diff bodies get really large, switch to a SAX parser to help with the memory footprint
         val bais = new ByteArrayInputStream(response.body)
         val gzis = new GZIPInputStream(bais)
+        val factory = SAXParserFactory.newInstance
+        val parser = factory.newSAXParser
+        val handler = new Change.ChangeHandler(sequence)
```

Comment:
Would it make sense to instantiate a single `SAXParser` and re-use it by `reset()`ing it after each parse?

(Same for the `ChangeHandler`?)

Reply:
I'm sorry, I was totally wrong about this. While testing, I discovered that `parse` is called multiple times concurrently.